### DISCUSSION TITLE
v20: default theme not using grid properly?

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [OpenMage/magento-lts#3964] [Ideas]
{{block type="catalog/product_list" name="home.catalog.product.list" alias="products_homepage" category_id="11" template="catalog/product/list.phtml"}}

![grid-layout](https://github.com/OpenMage/magento-lts/assets/13450219/43200304-7087-4abf-b5ac-83026ec5791d)

**Comment by pquerner** [comment]
Looks like some css didnt get applied, however I cannot confirm this from the demo store. Do you have some errors in the console?
https://demo.openmage.org/sale.html

**Comment by f1-outsourcing** [comment]
To be honest I was a bit in doubt if I was still on the 19 or on the newly 20.5.0 (I have both still running). Because my 20.5.0 has still the openmage logo in gif, your demo store has it in svg. 

My 20.5.0 is with the original 19 database, and currently merging of js and css is turned off. In the browser console network all files have a 200 
But these warnings are in the output.
```
Cookie “” has been rejected as third-party. css
Cookie “om_frontend” does not have a proper “SameSite” attribute value. Soon, cookies without the “SameSite” attribute or with an invalid value will be treated as “Lax”. This means that the cookie will no longer be sent in third-party contexts. If your application depends on this cookie being available in such contexts, please add the “SameSite=None“ attribute to it. To know more about the “SameSite“ attribute, read https://developer.mozilla.org/docs/Web/HTTP/Headers/Set-Cookie/SameSite spnl
Error in parsing value for '-webkit-text-size-adjust'.  Declaration dropped. styles.css:71:29
Error in parsing value for '-webkit-text-size-adjust'.  Declaration dropped. styles.css:379:29
Unrecognised at-rule or error parsing at-rule '@-ms-viewport'. styles.css:458:14
Unrecognised at-rule or error parsing at-rule '@-o-viewport'. styles.css:462:13
Unrecognised at-rule or error parsing at-rule '@viewport'. styles.css:466:10
Expected media feature name but found '-o-min-device-pixel-ratio'. styles.css:1115:80
Expected media feature name but found 'min-device-pixel-ratio'. styles.css:1115:116
Unknown property '-moz-border-radius'.  Declaration dropped. styles.css:2612:22
Unknown property '-moz-border-radius'.  Declaration dropped. styles.css:3799:22
Unknown property '-moz-border-radius'.  Declaration dropped. styles.css:3840:24
Expected media feature name but found '-o-min-device-pixel-ratio'. styles.css:4016:80
Expected media feature name but found 'min-device-pixel-ratio'. styles.css:4016:116
Unknown property '-moz-border-radius'.  Declaration dropped. styles.css:4066:22
Unknown pseudo-class or pseudo-element 'hidden'.  Ruleset ignored due to bad selector. styles.css:4255:35
Unknown pseudo-class or pseudo-element 'hidden'.  Ruleset ignored due to bad selector. styles.css:4626:21
Expected declaration but found ';'.  Skipped to next declaration. styles.css:5673:24
Unknown property 'zoom'.  Declaration dropped. styles.css:5674:8
Expected declaration but found ';'.  Skipped to next declaration. styles.css:5675:19
Unknown property 'zoom'.  Declaration dropped. styles.css:7489:8
Unknown property 'zoom'.  Declaration dropped. styles.css:7495:8
Unknown pseudo-class or pseudo-element '-ms-input-placeholder'.  Ruleset ignored due to bad selector. styles.css:8996:26
Unknown property '-moz-border-radius'.  Declaration dropped. styles.css:9510:22
Expected media feature name but found 'touch-enabled'. spnl:1:9
Expected media feature name but found '-webkit-touch-enabled'. spnl:1:25
Expected media feature name but found '-moz-touch-enabled'. spnl:1:49
Expected media feature name but found '-o-touch-enabled'. spnl:1:70
Expected media feature name but found '-ms-touch-enabled'. spnl:1:89
Expected media feature name but found 'modernizr'. spnl:1:109
Layout was forced before the page was fully loaded. If stylesheets are not yet loaded this may cause a flash of unstyled content. modernizr.custom.min.js:4:1969
```

**Comment by pquerner** [comment]
Something wierd going on with your database / installation. I noticed "SameSite" not having a proper value. That cant be the case as of 
https://github.com/OpenMage/magento-lts/pull/1246
It should be default to "None". Not saying thats THE issue with the css (obviously isnt), but it hints to more troubles.

Are you perhaps going back and forth with the same database on different codebases?

If you don't have anything custom installed (no modules), I'd check for things in the database. Maybe something is wierdly overwritten there.

**Reply by f1-outsourcing** [reply]
I will check this #1246

**Reply by f1-outsourcing** [reply]
I set this Same-Site to something else and then back to None, now this error is not listed any more.

**Reply by pquerner** [reply]
It has been long merged into the ^19 codebase. You'r 20.x whatever version should have that. It was one of the earliest. 
This one: https://github.com/OpenMage/magento-lts/releases/tag/v20.0.6
But its set on "" (empty string) in your case, which is very wierd.

**Reply by f1-outsourcing** [reply]
> Are you perhaps going back and forth with the same database on different codebases?

Yes I am comparing a bit, checking if everything is more or less the same. 

I didn't see any database upgrade scripts *.sql files in the source, so I thought maybe there were no changes.

**Reply by pquerner** [reply]
> I set this Same-Site to something else and then back to None, now this error is not listed any more.

Its fixed? Great! Wierd that it fixed it. Didnt expect it, hah! :D

**Reply by f1-outsourcing** [reply]
:) Yes this specific cookie message is gone, this layout is still the same.

**Reply by pquerner** [reply]
> :) Yes this specific cookie message is gone, this layout is still the same.

Oh. I thought the whole thing is fixed, not just the Cookie message. Well now I am sad again. ;D

**Reply by pquerner** [reply]
No idea then, sorry. Maybe you can set up a fresh database with the clean sample data, just to make 100% sure its not the core - its the data presented.

**Reply by f1-outsourcing** [reply]
did you check on the demo store (Or can I check) if it having issues when you create a page with this 'widget'

{{block type="catalog/product_list" name="home.catalog.product.list" alias="products_homepage" category_id="11" template="catalog/product/list.phtml"}}

Because if I have just a 'normal' storeview of a subcategory, everything also displays fine

**Reply by f1-outsourcing** [reply]
> set up a fresh database with the clean sample data

Is there css things stored in the database? Are there no individual sql update files with versions that I can apply or check? Quite  a bit of work starting from scratch.

**Reply by pquerner** [reply]
> Is there css things stored in the database? Are there no individual sql update files with versions that I can apply or check? Quite a bit of work starting from scratch.

Sure. CMS pages can have which sits in the database. I dont know your setup.
Also a clean database with M1 sample data installs within 30 seconds. Editing the .xml file takes another 30 seconds. Dont know what you mean by "quite a bit of work" exactly. ^^

**Comment by f1-outsourcing** [comment]
If I change the default product visibility from catalog,search to only catalog the product is not shown. So this is issue is maybe only related to search displays

**Comment by f1-outsourcing** [comment]
> Are you perhaps going back and forth with the same database on different codebases?

Yes I am comparing a bit, checking if everything is more or less the same. 

I didn't see any database upgrade scripts *.sql files in the source, so I thought maybe there were no changes.
